### ISSUE TITLE
Subscribe to new issues in a release

### ISSUE THREAD

**Issue by ryanwilsonperkin** [getsentry/sentry#15029]
## Summary

A user can subscribe to notifications of new issues that appear in a specific release.

## Motivation

We have (and I think this is a pretty common scenario) a project in Sentry with a large volume of errors in it. When we put out a new deployment, we often find ourselves watching the "New Issues" tab on the associated Sentry Release and refreshing for a little while. It would be great if we could instead tell the Release to notify us if there's a new issue that appears.

## Additional Context

I imagine a potential user experience for this could be a button on the release page that can be clicked to subscribe to, looking like this:

![image](https://user-images.githubusercontent.com/3004111/66531179-ea860d00-ead8-11e9-9442-40a3cf7f5f51.png)

It would also be useful to be able to subscribe a user programmatically with the sentry CLI, since we could then automatically subscribe people when they issue a deploy.

**Comment by adhiraj** [comment]
@ryanwilsonperkin we generally consider "new issue" to be an alert, not a workflow notification. Assuming that's what you want, do you watch for issues on every release, or only the occasional important ones? If the former, how would you expect this to behave differently from the "every new issue" alert rule we already have? If the latter, would you like to be unsubscribed from it after a while (to not keep receiving alerts from releases far in the past)?

**Comment by adhiraj** [comment]
Is this equivalent to creating an alert rule for new issues in release XYZ?
<img width="1112" alt="Screenshot 2019-10-31 23 12 29" src="https://user-images.githubusercontent.com/693121/68005894-0b530580-fc34-11e9-8b27-7efe33d10b83.png">
(Not suggesting that that's the optimal UX, just wondering if that's effectively what you're looking for. Also one limitation is that we don't currently have the ability to send alerts to specific users or teams)

**Comment by ryanwilsonperkin** [comment]
@adhiraj yes I think that would be functionally equivalent with the one exception that we _would_ want it to be a notification for a specific user or team. Without the send the notification specifically, it falls into the same shortcomings as a more general rule for "notify on all new issues".

**Comment by BYK** [comment]
Closing this issue due to staleness. Feel free to comment here if you think we should still work on this.

**Comment by adhiraj** [comment]
a) do we need a personal subscription (i.e. any user who subscribes gets notifications), or to be able to specify a user or team that new issues in the release should go to? b) do we need to be able to send this to integrations (e.g. slack)?

**Comment by ryanwilsonperkin** [comment]
> a) do we need a personal subscription (i.e. any user who subscribes gets notifications), or to be able to specify a user or team that new issues in the release should go to? 

Both would likely be desirable. We'd want the ability to opt people in as part of our CD pipeline, but others might also choose to opt in when visiting the release page.

> b) do we need to be able to send this to integrations (e.g. slack)?

Doesn't personally matter to me, since we'd handle that separately either way.


I should note that I no longer work for the employer who I initially opened this on behalf of, and my new role doesn't have me using sentry. That said, I'd still advocate for this feature because it seems useful (and I want to use sentry again in the future 😉)

**Comment by snigdhas** [comment]
We've attempted to build some version of release notifications a few times, and I agree there's still a good argument for the value of this sort of feature. Would be a great topic for a brainstorming session.
